src/reference/snakemake-style-guide.rst

```diff
@@ -193,3 +193,36 @@ Example:
            --output-sequences {output.sequences:q} \
            --output-metadata {output.metadata:q}
        """
+
+Log standard out and error output to log files and the terminal
+===============================================================
+
+Use `the Snakemake log directive <https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#log-files>`_ for each rule that writes output to either standard out or error and direct output to the corresponding log file.
+Use the ``tee`` command to ensure that output gets written to both the log file and the terminal, so users can track their workflow progress interactively and use the log file later for debugging.
+
+Example:
+
+.. code-block:: python
+
+   rule filter:
+       input:
+           metadata="results/metadata.tsv",
+       output:
+           metadata="results/filtered_metadata.tsv",
+       log:
+           "logs/filter.txt"
+       shell:
+           """
+           augur filter \
+               --metadata {input.metadata} \
+               --output-metadata {output.metadata} 2>&1 | tee {log}
+           """
+
+Before using ``tee``, ensure that your workflow uses bash's ``pipefail`` option, so successful ``tee`` execution does not mask errors from earlier commands in the pipe.
```

Comment:
We should document how to do this (I don't see it in this style guide)

Reply:
Good call. Since the pipefail option is enabled by default in Snakemake, we shouldn't have to do anything but check for parts of the workflow that override the defaults. I added examples of how to check that the defaults are overridden and also a link to the definition of the "bash strict mode".